samples/Mvc.Client/Controllers/AuthenticationController.cs

```diff
@@ -35,7 +40,31 @@ public async Task<IActionResult> SignIn([FromForm] string provider)
             // Instruct the middleware corresponding to the requested external identity
             // provider to redirect the user agent to its own authorization endpoint.
             // Note: the authenticationScheme parameter must match the value configured in Startup.cs
-            return Challenge(new AuthenticationProperties { RedirectUri = "/" }, provider);
+            if (provider != ShopifyAuthenticationDefaults.AuthenticationScheme)
+            {
+                return Challenge(new AuthenticationProperties
+                {
+                    RedirectUri = "/",
+                }, provider);
+            }
+
+
+            // Shopify OAuth differs from most (all?) others in that you need to know the host name of the
+            // shop in order to construct the authorization endpoint. This can be aquired either from the
+            // user directly, or provided by the shopify app store during app install/activation.
+            var authProps = new ShopifyAuthenticationProperties("ctest-205") // Put your shop name here.
```

Comment:
Could you put all the Shopify-specific code into its own method and do _if Shopify (do other thing)_? Otherwise the code reads as it's mainly for Shopify, rather than the other way around.

Reply:
Hey Martin, sorry to have let this sit so long. I was diagnosed with leukemia about 3 days after I submitted the PR. I'm still only working half time and have about a year's worth of backlogged tasks. I really appreciate all your comments! I'll try to find some time for this asap.

Reply:
No problem, Chris - I've only recently become a maintainer myself, and I went through all the PRs and issues to either push them forwards or close them off.

A future update to re-implement this would be great if you manage find the time 😃

Reply:
> I was diagnosed with leukemia about 3 days after I submitted the PR.

@ChrisInSeattle I'm really sorry to hear that. I wish you a speedy recovery. Take care of you! 🤞